zhusuan/distributions/multivariate.py

```diff
@@ -29,6 +33,158 @@
+    The class of multivariate normal distribution, where covariance is
+    parameterized with the lower triangular matrix in Cholesky decomposition,
+
+        .. math :: L \\text{s.t.} LL^T = \Sigma.
+
+    See :class:`~zhusuan.distributions.base.Distribution` for details.
+
+    :param mean: An N-D `float` Tensor. mean[..., i_{N-1}, :] corresponds to a
+        single distribution.
+    :param cov_tril: An (N+1)-D `float` Tensor. Each slice represents the lower
+        triangular matrix in the Cholesky decomposition.
+    :param group_ndims: A 0-D `int32` Tensor representing the number of
+        dimensions in `batch_shape` (counted from the end) that are grouped
+        into a single event, so that their probabilities are calculated
+        together. Default is 0, which means a single value is an event.
+        See :class:`~zhusuan.distributions.base.Distribution` for more detailed
+        explanation.
+    :param is_reparameterized: A Bool. If True, gradients on samples from this
+        distribution are allowed to propagate into inputs, using the
+        reparametrization trick from (Kingma, 2013).
+    :param use_path_derivative: A bool. Whether when taking the gradients
+        of the log-probability to propagate them through the parameters
+        of the distribution (False meaning you do propagate them). This
+        is based on the paper "Sticking the Landing: Simple,
+        Lower-Variance Gradient Estimators for Variational Inference"
+    :param check_numerics: Bool. Whether to check numeric issues.
+    """
+
+    def __init__(self,
+                 mean,
+                 cov_tril,
+                 group_ndims=0,
+                 is_reparameterized=True,
+                 use_path_derivative=False,
+                 check_numerics=False,
+                 **kwargs):
+        self._check_numerics = check_numerics
+        self._mean = assert_rank_at_least_one(mean, 'MVNTriL.mean')
+        self._n_dim = get_shape_at(self._mean, -1)
+        self._cov_tril = assert_rank_at_least(cov_tril, 2, 'MVNTriL.cov_tril')
+
+        # Check shape; fail asap
+        if self._mean.get_shape() and self._cov_tril.get_shape():
+            expected_shape = get_shape_list(self._mean) + [self._n_dim]
+            actual_shape = get_shape_list(self._cov_tril)
+            err_msg = ['MVNTriL.cov_tril should have compatible shape with',
+                       ' mean. Expected', tf.convert_to_tensor(expected_shape),
+                       'got', tf.convert_to_tensor(actual_shape)]
+            assert_ops = [tf.assert_equal(x, y, err_msg)
+                          for x, y in zip(expected_shape, actual_shape)]
+        else:
+            expected_shape = tf.concat(
+                    [tf.shape(self._mean), [self._n_dim]], axis=0)
+            actual_shape = tf.shape(self._cov_tril)
+            msg = ['MVNTriL.cov_tril should have compatible shape with '
+                   'mean. Expected', expected_shape, ' got ', actual_shape]
```

Comment:
I don't see documents on this type of usage. Where did you find it?

Reply:
https://www.tensorflow.org/api_docs/python/tf/assert_equal

Reply:
But they did not say the message can be a list?

Reply:
It's the `data` argument, a list of Tensors.